ev3dev/core.py

```diff
@@ -2171,19 +2173,70 @@ def trigger(self):
     def trigger(self, value):
         self._trigger = self.set_attr_string(self._trigger, 'trigger', value)
 
+        # Workaround for ev3dev/ev3dev#225.
+        # When trigger is set to 'timer', we need to wait for 'delay_on' and
+        # 'delay_off' attributes to appear with correct permissions.
+        if value == 'timer':
+            for attr in ('delay_on', 'delay_off'):
+                path = self._path + '/' + attr
+
+                # Make sure the file has been created:
+                for _ in range(5):
+                    if os.path.exists(path):
+                        break
+                    time.sleep(0.2)
+                else:
+                    raise Exception('"{}" attribute has not been created'.format(attr))
+
+                # Make sure the file has correct permissions:
+                for _ in range(5):
+                    mode = stat.S_IMODE(os.stat(path)[stat.ST_MODE])
+                    if mode & stat.S_IRGRP and mode & stat.S_IWGRP:
+                        break
+                    time.sleep(0.2)
+                else:
+                    raise Exception('"{}" attribute has wrong permissions'.format(attr))
+
+
     @property
     def delay_on(self):
         """
         The `timer` trigger will periodically change the LED brightness between
         0 and the current brightness setting. The `on` time can
         be specified via `delay_on` attribute in milliseconds.
         """
-        self._delay_on, value = self.get_attr_int(self._delay_on, 'delay_on')
-        return value
+
+        # Workaround for ev3dev/ev3dev#225.
+        # 'delay_on' and 'delay_off' attributes are created when trigger is set
+        # to 'timer', and destroyed when it is set to anything else.
+        # This means the file cache may become outdated, and we may have to
+        # reopen the file.
+        for i in range(2):
```

Comment:
I guess this could be turned into a `whileTrue`, but then I would need a boolean variable to distinguish the first run from the second. And `i` does that for me already.